rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/history/HistoryStore.scala

```diff
@@ -17,24 +17,27 @@ object HistoryStoreInstances {
   def historyStore[F[_]: Sync](store: Store[F]): HistoryStore[F] = new HistoryStore[F] {
     // TODO put list
     override def put(tries: List[Trie]): F[Unit] = {
-      val data = tries
-        .map { t =>
-          val key   = Trie.hash(t)
-          val bytes = Trie.codecTrie.encode(t).get
-          (key, bytes)
-        }
-      store.put(data)
+      def put(t: Trie): F[Unit] = {
+        val k = Trie.hash(t)
+        for {
+          b <- Trie.codecTrie.encode(t).get
+          _ <- store.put(k, b)
```

Comment:
the whole idea for `put(List)` is to have a bulk action in one low level txn. If introducing F in Attempt forces us to not have this opportunity - I vote against this change.

Reply:
tnx, @dzajkowski fixed the issue in the new commit